packages/codecs-data-structures/src/__tests__/enum-helpers-test.ts

```diff
@@ -0,0 +1,224 @@
+import {
+    formatNumericalValues,
+    getEnumIndexFromDiscriminator,
+    getEnumIndexFromVariant,
+    getEnumStats,
+} from '../enum-helpers';
+
+describe('getEnumStats', () => {
+    it('handles sequential numerical enums', () => {
+        enum Feedback {
+            Bad,
+            Good,
+        }
+        expect(getEnumStats(Feedback)).toStrictEqual({
+            enumKeys: ['Bad', 'Good'],
+            enumRecord: { Bad: 0, Good: 1 },
+            enumValues: [0, 1],
+            numericalValues: [0, 1],
+            stringValues: ['Bad', 'Good'],
+        });
+    });
+
+    it('handles explicit numerical enums', () => {
+        enum Prime {
+            Two = 2,
+            Three,
+            Five = 5,
+        }
+        expect(getEnumStats(Prime)).toStrictEqual({
+            enumKeys: ['Two', 'Three', 'Five'],
+            enumRecord: { Five: 5, Three: 3, Two: 2 },
+            enumValues: [2, 3, 5],
+            numericalValues: [2, 3, 5],
+            stringValues: ['Two', 'Three', 'Five'],
+        });
+    });
+
+    it('handles conflicting numerical enums', () => {
+        enum FortyTwo {
+            One = 42,
+            // eslint-disable-next-line @typescript-eslint/no-duplicate-enum-values
+            Two = 42,
+        }
```

Comment:
Ahh so at least ESLint will barf on this (thank god).

Reply:
Haha yeah I was pretty pleased to see that too.